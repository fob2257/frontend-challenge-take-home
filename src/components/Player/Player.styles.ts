import styled, { css } from 'styled-components';

export const Container = styled.div`
  width: 560px;
  height: 132px;
  background: #f5f5f5;

  display: flex;
  align-items: center;

  @media (max-width: 1024px) {
    width: 100%;
    position: relative;
  }
`;

export const GrowDiv = styled.div`
  flex-grow: 1;
`;

export const PaddedDiv = styled.div<any>`
  padding: 0 16px;

  @media (max-width: 1024px) {
    ${({ absolute }) => {
      if (absolute) {
        return css`
          padding: 0 35px 0 12px;
          position: absolute;
          padding: 0;
          left: 86px;
        `;
      }

      return css`
        padding: 0 35px 0 12px;
      `;
    }}
  }
`;

export const Img = styled.img`
  width: 100px;
  height: 100px;
`;

const fontStyles = css`
  font-family: 'Roboto', sans-serif;
  font-style: normal;
  color: #000000;
  font-weight: 700;
  font-size: 20px;

  margin: 0;
`;

export const Title = styled.h3`
  ${fontStyles}
  margin-bottom: 8px;

  @media (max-width: 320px) {
    font-size: 16px;
  }
`;

export const Data = styled.p`
  ${fontStyles}
  font-weight: 400;
  font-size: 16px;
  margin-bottom: 8px;

  @media (max-width: 320px) {
    font-size: 12px;
  }
`;
