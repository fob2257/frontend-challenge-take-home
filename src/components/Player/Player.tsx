import React from 'react';

import {
  Container,
  GrowDiv,
  PaddedDiv,
  Img,
  Title,
  Data,
} from './Player.styles';
import Ratio from '../Ratio/Ratio';

import { Player } from '../../data/types';
import { getLastPlayedDate, getPlayerWinRatio } from '../../utils/utils';

type Props = {
  player: Player;
};

const PlayerComponent: React.FC<Props> = ({ player }) => {
  const lastActivity = getLastPlayedDate(player);
  const opponents = player.games.filter(
    (game) => game.winnerName !== player.fullName,
  );
  const ratio = getPlayerWinRatio(player);

  return (
    <Container data-testid="player-card">
      <PaddedDiv>
        <Img src={player.avatar} alt="Player avatar" />
      </PaddedDiv>
      <GrowDiv>
        <Title>{player.fullName}</Title>
        <Data>Last Activity: {lastActivity}</Data>
        <Data>Unique Opponents: {opponents.length}</Data>
      </GrowDiv>
      <PaddedDiv absolute>
        <Ratio ratio={ratio} />
      </PaddedDiv>
    </Container>
  );
};

export default PlayerComponent;
