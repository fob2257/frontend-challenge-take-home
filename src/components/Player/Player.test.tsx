import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import Player from './Player';

import { players } from '../../data/fakeData';

describe('<Player />', () => {
  const setup = (name?: string) => {
    const [player] = players;

    if (name) {
      player.fullName = name;
    }

    return <Player player={player} />;
  };

  it('should render expected value', () => {
    const name = 'Ron Swanson';

    render(setup(name));

    const element = screen.getByText(new RegExp(name, 'i'));

    expect(element).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const component = create(setup());
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
