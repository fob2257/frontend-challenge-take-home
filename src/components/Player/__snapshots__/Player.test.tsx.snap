// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> should match snapshot 1`] = `
<div
  className="sc-bczRLJ giaMah"
  data-testid="player-card"
>
  <div
    className="sc-dkzDqf gXsSwv"
  >
    <img
      alt="Player avatar"
      className="sc-hKMtZM glFvRz"
      src="https://www.adoptapet.com/blog/uploads/2013/01/Mango-cat-collar.jpg"
    />
  </div>
  <div
    className="sc-gsnTZi fcSHXD"
  >
    <h3
      className="sc-eCYdqJ itSnir"
    >
      Ron Swanson
    </h3>
    <p
      className="sc-jSMfEi jHockD"
    >
      Last Activity: 
      2020-02-20
    </p>
    <p
      className="sc-jSMfEi jHockD"
    >
      Unique Opponents: 
      3
    </p>
  </div>
  <div
    className="sc-dkzDqf cwMyMC"
  >
    <div
      className="sc-gKXOVf mRcIl"
    >
      <p
        className="sc-iBkjds frpA-dj"
      >
        0
      </p>
    </div>
  </div>
</div>
`;
